### DISCUSSION TITLE
Why does Colors.cs use a Dictionary for storing the colors?

### DISCUSSION THREAD

**Discussion by c-schembri** [godotengine/godot-proposals#5202] [Scripting]
I'm talking about the `Colors.cs` file found at `modules/mono/glue/GodotSharp/GodotSharp/Core/Colors.cs`.

This file has a `Dictionary<string, Color>` which stores a list of colors. Accessing any of these colors is then done like so: 
`public static Color AliceBlue { get { return namedColors["ALICEBLUE"]; } }`

Why is the implementation not just done like the following?
`public static readonly Color AliceBlue = new Color(0.94f, 0.97f, 1.00f);`

What's the need for the dictionary when we have a getter for every dictionary entry anyway?

**Comment by Mickeon** [comment]
I don't know much about C# in Godot, but the `Color()` constructor can also accept a String that matches one of those colours, and returns it as a result.

**Reply by c-schembri** [reply]
Ah yeah, you're right (kind of). There is no constructor in the C# code but there is a function in the Color struct like so:
`public static Color ColorN(string name, float alpha = 1f)`

That's a weird name for the function but this nevertheless justifies the use of a dictionary. Though I'm not sure why this isn't just another ctor for the Color struct.
